### PR TITLE
[flutter_markdown] Added option selectionControls

### PR DESCRIPTION
**Pull request by spatster**
Added option selectionControls.

flutter_markdown uses SelectableText for markdown rendering. It has selectionControls parameter for creating a custom context menu.  This PR added this parameter to flutter_markdown and passes it to SelectableText widget.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Thanks for the submission!

> [x] I listed at least one issue that this PR fixes in the description above.

I don't see this in the PR description. Could you add a link to an issue that describes the goal and use case here? There's not much context in the PR description currently.

> [x] All existing and new tests are passing.

Please fix the license issue that's flagged by the `format+analyze` CI step, so that all of the test steps will run.

**Comment by spatster** [comment]
> Thanks for the submission!
> 
> > [x] I listed at least one issue that this PR fixes in the description above.
> 
> I don't see this in the PR description. Could you add a link to an issue that describes the goal and use case here? There's not much context in the PR description currently.
> 
> > [x] All existing and new tests are passing.
> 
> Please fix the license issue that's flagged by the `format+analyze` CI step, so that all of the test steps will run.

done

File: packages/flutter_markdown/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.7.0
+
+  * Added option selectionControls
```
Comment (stuartmorgan) [inline packages/flutter_markdown/CHANGELOG.md:3]: Nit: Adds

See the CHANGELOG style guide linked from the PR checklist.

File: packages/flutter_markdown/test/text_selection_controls.dart [hunk 2]
```diff
@@ -0,0 +1,316 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+const double _kHandleSize = 22.0;
```
Comment (stuartmorgan) [inline packages/flutter_markdown/test/text_selection_controls.dart:10]: Dart constants should not use `k`, per Dart style.

File: packages/flutter_markdown/test/text_selection_controls.dart [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+// Padding between the toolbar and the anchor.
+const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
+const double _kToolbarContentDistance = 8.0;
+
+/// Android Material styled text selection controls.
+class MaterialTextSelectionControls extends TextSelectionControls {
```
Comment (stuartmorgan) [inline packages/flutter_markdown/test/text_selection_controls.dart:17]: Why does this class need to be so elaborate when all its used for is unit tests? I'm not seeing how all of the drawing details here affect the unit test.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
Adding @domesticmouse for primary review.

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (spatster) [reply packages/flutter_markdown/CHANGELOG.md:3]: I didn't understand how to change. I use the current style.

## 0.6.8

  * Added option paddingBuilders
  
## 0.6.7

 * Fix `unnecessary_import` lint errors.
 * Added option pPadding
 * Added options h1Padding - h6Padding

File: packages/flutter_markdown/test/text_selection_controls.dart (hunk 2 above)
Comment (spatster) [reply packages/flutter_markdown/test/text_selection_controls.dart]: Updated.

File: packages/flutter_markdown/test/text_selection_controls.dart (hunk 3 above)
Comment (spatster) [reply packages/flutter_markdown/test/text_selection_controls.dart:17]: I took the test from https://github.com/flutter/flutter/blob/33e261c0a2eb1f02b94c8fdfef092350c65b4d5c/packages/flutter/lib/src/material/text_selection.dart

Maybe add information about it to the test header?

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/flutter_markdown/CHANGELOG.md:3]: > I didn't understand how to change.

You need to change "Added" to "Adds"

> I use the current style.

You need to follow the store guide I referenced, which is newer than the existing entries.

File: packages/flutter_markdown/test/text_selection_controls.dart (hunk 3 above)
Comment (stuartmorgan) [reply packages/flutter_markdown/test/text_selection_controls.dart:17]: This doesn't answer my question. Why do you need an elaborate custom selection control system, with lots of code for things like the exact drawing of the selection handle, when your test doesn't set anything about those details?

File: packages/flutter_markdown/CHANGELOG.md (hunk 1 above)
Comment (spatster) [reply packages/flutter_markdown/CHANGELOG.md]: Thanks. Updated

**Review by domesticmouse — COMMENTED** [review]
Can you also answer @stuartmorgan's question please?

**Pull request closed without merging**